test/token-service/test_tokenUpdateTransaction.js

```diff
@@ -0,0 +1,3397 @@
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+  });
+
+  describe("Expiration Time", function () {
+    it.skip("(#1) Updates an immutable token with a valid expiration time", async function () {
+      const expirationTime = (
+        Math.floor(Date.now() / 1000) + 5184000
+      ).toString();
+
+      try {
+        await JSONRPCRequest(this, "updateToken", {
+          tokenId: immutableTokenId,
+          expirationTime,
+        });
+      } catch (err) {
+        assert.equal(err.data.status, "TOKEN_IS_IMMUTABLE");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#2) Updates a mutable token to an expiration time of 0", async function () {
+      try {
+        await JSONRPCRequest(this, "updateToken", {
+          tokenId: mutableTokenId,
+          expirationTime: "0",
+          commonTransactionParams: {
+            signers: [mutableTokenKey],
+          },
+        });
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#3) Updates a mutable token to an expiration time of -1", async function () {
+      try {
+        await JSONRPCRequest(this, "updateToken", {
+          tokenId: mutableTokenId,
+          expirationTime: "-1",
+          commonTransactionParams: {
+            signers: [mutableTokenKey],
+          },
+        });
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#4) Updates a mutable token to an expiration time of 9,223,372,036,854,775,807 (int64 max) seconds", async function () {
+      try {
+        await JSONRPCRequest(this, "updateToken", {
+          tokenId: mutableTokenId,
+          expirationTime: "9223372036854775807",
+          commonTransactionParams: {
+            signers: [mutableTokenKey],
+          },
+        });
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#5) Updates a mutable token to an expiration time of 9,223,372,036,854,775,806 (int64 max - 1) seconds", async function () {
+      try {
+        await JSONRPCRequest(this, "updateToken", {
+          tokenId: mutableTokenId,
+          expirationTime: "9223372036854775806",
+          commonTransactionParams: {
+            signers: [mutableTokenKey],
+          },
+        });
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_EXPIRATION_TIME");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#6) Updates a mutable token to an expiration time of -9,223,372,036,854,775,808 (int64 min) seconds", async function () {
```

Comment:
For me this test is failing, I debugged in services and the expiry gets ignored, thus making this method not throw an error as it should.

Here is the relevant snippet from services
![image](https://github.com/user-attachments/assets/dc27478d-031b-4f2b-94ea-66f32bd0e407)

Reply:
Yes, it should be skipped. I wasn't passing the correct value as a property to `setExpirationTime` because I was creating a new Date object from a large "number" casted to an integer, which resulted in an "Invalid Date" output. After fixing it by creating a new timestamp from a string cast to a long type, it worked correctly. This is a bug in the JS TCK server.

`new Timestamp(Long.fromString(expirationTime), 0)` -> this is how I am casting it now.

As a result, this test fails as well. On my side, this update should also be applied in tokenCreate. I suspect the same issue exists with the C++ server.

Reply:
This looks to be the same issue we had with `autoRenewPeriod`, setting the `int64` min value would appear as unset in services and cause this. Agreed that it can be skipped and we can let services know.